database/postgresconn/postgresconn.go

```diff
@@ -294,24 +292,46 @@ func runesLastIndex(input []rune, target rune) int {
 }
 
 func (p *Postgres) SetVersion(version int, dirty bool) error {
-	tx, err := p.conn.BeginTx(context.Background(), &sql.TxOptions{})
+	ctx := context.Background()
+	tx, err := p.conn.BeginTx(ctx, &sql.TxOptions{})
 	if err != nil {
 		return &database.Error{OrigErr: err, Err: "transaction start failed"}
 	}
 
-	// Also re-write the schema version for nil dirty versions to prevent
-	// empty schema version for failed down migration on the first migration
-	// See: https://github.com/getoutreach/migrate/issues/330
-	query := fmt.Sprintf(`INSERT INTO %q.%q`+
-		` (version, dirty, created_at) VALUES ($1, $2, now())`,
-		p.config.migrationsSchemaName, p.config.migrationsTableName)
-	if _, err := tx.Exec(query, version, dirty); err != nil {
-		if errRollback := tx.Rollback(); errRollback != nil {
-			err = multierror.Append(err, errRollback)
+	// check for in progress version, if it exists use the in-progress
+	// version to record the dirty, info etc. values.
+	row := tx.QueryRowContext(ctx,
+		fmt.Sprintf(
+			`SELECT id FROM %q.%q WHERE version = $1 ORDER BY created_at DESC limit 1`,
```

Comment:
Make sense to have a uniqueness constraint on version?

Reply:
Added